draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -274,70 +291,176 @@ The array is created as follows:
+### Array Elements
 
-The `_sd` key MUST refer to an array of strings, each string being a digest of a Disclosure or a decoy digest as described above.
+Digests of Disclosures for array elements are added to the array in the same
+position as the original claim value in the array. For each digest, an object
+of the form `{"...": "<digest>"}` is added to the array. The key MUST be the
+string `...` (three dots). The value MUST be the digest of the Disclosure as
+described in (#hashing_disclosures). There MUST NOT be any other keys in the
+object.
 
-The array MAY be empty in case the Issuer decided not to selectively disclose any of the claims at that level. However, it is RECOMMENDED to omit the `_sd` key in this case to save space.
+For example, using the digest of the array element Disclosure created above,
+the Issuer could create the following SD-JWT payload to make the second element
+of the `nationalities` array selectively disclosable:
 
-The Issuer MUST hide the original order of the claims in the array. To ensure this, it is RECOMMENDED to shuffle the array of hashes, e.g., by sorting it alphanumerically or randomly. The precise method does not matter as long as it does not depend on the original order of elements.
+```json
+{
+  "nationalities":
+    ["DE", {"...": "w0I8EKcdCtUPkGCNUrfwVp2xEgNjtoIDlOxc9-PlOhs"}]
+}
+```
 
-Issuers MUST NOT issue SD-JWTs where
+As described in (#verifier_verification), Verifiers ignore all selectively
+disclosable array elements for which they did not receive a Disclosure. In the
+example above, the verification process would output an array with only one
+element unless a matching Disclosure for the second element is received.
 
- * the key `_sd` is already used for the purpose other than to contain the array of digests, or
- * the same Disclosure value appears more than once (in the same array or in different arrays).
+## Example 1: SD-JWT {#example-1}
 
+This example uses the following object as the set of claims that the Issuer is issuing:
+
+<{{examples/simple/user_claims.json}}
+
+The following non-normative example shows a payload of an SD-JWT for this End-User data:
+
+<{{examples/simple/sd_jwt_payload.json}}
+
+The Issuer in this case made the following decisions:
+
+* The `nationalities` array is always visible, but its contents are selectively disclosable.
+* The `sub` element and essential verification data (`iss`, `iat`, `cnf`, etc.) are always visible.
+* All other End-User claims are selectively disclosable.
+* For `address`, the Issuer is using a flat structure, i.e., all of the claims
+  in the `address` claim can only be disclosed in full. Other options are
+  discussed in (#nested_data).
+
+The Issuer creates the following Disclosures:
+
+{{examples/simple/disclosures.md}}
+
+The SD-JWT is then signed by the Issuer to create a JWT like the following:
+
+<{{examples/simple/sd_jwt_serialized.txt}}
+
+## Decoy Digests {#decoy_digests}
```

Comment:
```suggestion
## Adding Decoy Digests {#decoy_digests}
```

Reply:
Since this explains decoy digests (and not only how to add them), I'd like to keep the title as is.

Reply:
ok